liquidctl/driver/asetek_pro.py

```diff
@@ -0,0 +1,271 @@
+        msg = self._end_transaction_and_read(4)
+        #LOGGER.debug('pump mode msg %s', ' '.join(format(i, '02x') for i in msg))
+        pump_mode = msg[3]
+        status.append(('Pump mode', _PRO_PUMP_MODES[pump_mode], ""))
+        self._write([_PRO_CMD_READ_PUMP_SPEED])
+        msg = self._end_transaction_and_read(5)
+        #LOGGER.debug('pump speed msg %s', ' '.join(format(i, '02x') for i in msg))
+        pump_speed = (msg[3] << 8) + msg[4]
+        status.append(('Pump speed', pump_speed | 0, 'rpm'))
+        self._write([_PRO_CMD_READ_FIRMWARE])
+        msg = self._end_transaction_and_read(7)
+        #LOGGER.debug('firmware id msg %s', ' '.join(format(i, '02x') for i in msg))
+        firmware = '{}.{}.{}.{}'.format(format(msg[3], '01x'), format(msg[4], '01x'),
+            format(msg[5], '01x'),format(msg[6], '01x'))
+        status.append(('Firmware version', firmware, ''))
+        return status
+
+    def set_color(self, channel, mode, colors, time_per_color=1, time_off=None,
+                  alert_threshold=_HIGH_TEMPERATURE, alert_color=[255, 0, 0],
+                  speed=3, temps=['35', '45', '55'], **kwargs):
+        """Set the color mode for a specific channel."""
+        LOGGER.debug('kwargs %s', kwargs)
+        colors = list(colors)
+        LOGGER.debug('color count = %d', len(colors))
+        LOGGER.debug('speed %s', int(speed))
+        if mode == 'rainbow':
+            self._write_color_change_speed(mode, speed)
+            self._write([0x55, 0x01])
+            self._end_transaction_and_read(3)
+        elif mode == 'shift':
+            self._write_color(colors)
+            self._write_color_change_speed(mode, speed)
+            self._write([0x55, 0x01])
+            self._end_transaction_and_read(3)
+        elif mode == 'pulse':
+            self._write_color(colors)
+            self._write_color_change_speed(mode, speed)
+            self._write([0x52, 0x01])
+            self._end_transaction_and_read(3)
+        elif mode == 'blinking':
+            self._write_color(colors)
+            self._write_color_change_speed(mode, speed)
+            self._write([0x58, 0x01])
+            self._end_transaction_and_read(3)
+        elif mode == 'fixed':
+            self._write_color([colors[0], colors[0]])
+            self._write([0x55, 0x01])
+            self._end_transaction_and_read(3)
+        elif mode == 'alert':
+            temps = list(temps.split(' '))
+            LOGGER.debug('temps count = %d', len(temps))
+            LOGGER.debug('asetek temps %s %s %s', temps[0], temps[1], temps[2])
+            LOGGER.debug('color count = %d', len(colors))
+            self._write([0x5f, int(temps[0]), 0x00, int(temps[1]), 0x00, int(temps[2]), 0x00] + colors[0] + colors[1] + colors[2])
+            self._end_transaction_and_read(6)
+            self._write([0x5E, 0x01])
+            self._end_transaction_and_read(3)
+        else:
+            raise KeyError('Unknown lighting mode {}'.format(mode))
```

Comment:
Can this be refactored to have less duplicate code?

Reply:
Done